tests/clients/test_pep0249.py

```diff
@@ -0,0 +1,320 @@
+            assert sample_params in response
+            assert len(response) == 2
+
+    def test_callproc_with_tracing(self):
+        with tracer.start_as_current_span("test"):
+            sample_proc_name = "call insert_user(%s, %s, %s);"
+            sample_params = (9, "sample-name-9", "sample-email-9@mail.com")
+            self.test_wrapper.callproc(sample_proc_name, sample_params)
+            self.test_conn.commit()
+            self.test_wrapper.execute("select * from tests;")
+            response = self.test_wrapper.fetchall()
+            assert sample_params in response
+            assert len(response) == 2
+
+            # Exception Handling
+            error_proc_name = "erroroeus command;"
+            with pytest.raises(Exception) as exc_info, patch.object(
+                InstanaSpan,
+                "record_exception",
+            ) as mock_exception:
+                self.test_wrapper.callproc(error_proc_name, sample_params)
+            assert exc_info.typename == "SyntaxError"
+            mock_exception.call_count == 2
+
+
+class TestConnectionWrapper:
+    @pytest.fixture(autouse=True)
+    def _setup(self) -> Generator[None, None, None]:
+        self.connect_params = [
+            "db",
+            {
+                "db": "instana_test_db",
+                "host": "localhost",
+                "port": "5432",
+                "user": "root",
+                "password": "passw0rd",
+            },
+        ]
+        self.test_conn = psycopg2.connect(
+            database=self.connect_params[1]["db"],
+            host=self.connect_params[1]["host"],
+            port=self.connect_params[1]["port"],
+            user=self.connect_params[1]["user"],
+            password=self.connect_params[1]["password"],
+        )
+        self.module_name = "test-connection"
+        self.connection_manager = ConnectionWrapper(
+            self.test_conn, self.module_name, self.connect_params
+        )
+        yield
+        self.test_conn.close()
+
+    def test_enter(self):
+        response = self.connection_manager.__enter__()
+        assert isinstance(response, ConnectionWrapper)
+        assert response._module_name == self.module_name
+        assert response._connect_params == self.connect_params
+
+    def test_cursor(self):
+        response = self.connection_manager.cursor()
+        assert isinstance(response, CursorWrapper)
+
+    def test_close(self):
+        response = self.connection_manager.close()
+        assert self.test_conn.closed
+        assert not response
+
+    def test_commit(self):
+        response = self.connection_manager.commit()
+        assert not response
+
+    def test_rollback(self):
+        if hasattr(self.connection_manager, "rollback"):
+            response = self.connection_manager.rollback()
+            assert not response
+
+
+class TestConnectionFactory:
+    @pytest.fixture(autouse=True)
+    def _setup(self) -> Generator[None, None, None]:
```

Comment:
Same as [this](https://github.com/instana/python-sensor/pull/575/files#r1713859847) comment.